src/drake/parser.clj

```diff
@@ -670,19 +682,22 @@
     _ (p/opt inline-comment)
     vars (p/get-info :vars)
     methods (p/get-info :methods)
-    _ (p/failpoint line-break (illegal-syntax-error-fn "%call / %include"))]
+    _ (p/failpoint line-break (illegal-syntax-error-fn "%call / %include / %context"))]
    (let [raw-base (get vars "BASE" default-base)
          base (add-path-sep-suffix raw-base)
          ;; Need to use fs/file here to honor cwd
          ^String tokens (slurp (fs/file file-path))
+         exec-dir (when (= directive directive-context) (dfs/get-directory-path file-path))
          prod (parse-state
                (assoc (make-state
                         (ensure-final-newline tokens)
                         vars methods 0 0)
+                 :exec-dir exec-dir
```

Comment:
I don't really like passing `:exec-dir nil` when there is no exec-dir, compared to not passing those two at all. Someone downstream might do something like `(get opt-map :exec-dir *cwd*)`, which will return `nil` in your formulation, rather than `*cwd*` as they surely intend.